frame/executive/src/lib.rs

```diff
@@ -260,42 +260,44 @@ where
 	/// hashes.
 	pub fn apply_extrinsic(uxt: Block::Extrinsic) -> ApplyExtrinsicResult {
 		let encoded = uxt.encode();
-		let encoded_len = encoded.len();
-		Self::apply_extrinsic_with_len(uxt, encoded_len, Some(encoded))
+		Self::apply_extrinsic_with_len(uxt, &encoded, true)
 	}
 
 	/// Apply an extrinsic inside the block execution function.
 	fn apply_extrinsic_no_note(uxt: Block::Extrinsic) {
-		let l = uxt.encode().len();
-		match Self::apply_extrinsic_with_len(uxt, l, None) {
+		let encoded = uxt.encode();
+		match Self::apply_extrinsic_with_len(uxt, &encoded, false) {
 			Ok(_) => (),
 			Err(e) => { let err: &'static str = e.into(); panic!(err) },
 		}
 	}
 
-	/// Actually apply an extrinsic given its `encoded_len`; this doesn't note its hash.
+	/// Actually apply an extrinsic given its `encoded_len`
```

Comment:
I am having a hard time following: 

What will be placed in `ExtrinsicData ` in system, and when? 
What will be placed in `EXTRINSIC_DATA ` directly, and when?

(to me it seems like they are duplicate and the latter should be a `EXTRINSIC_HASH` but I want to pose it as a question). 

And whatever is the conclusion should be documented in the function's header.